lib/spdy-transport/connection.js

```diff
@@ -281,7 +281,7 @@ Connection.prototype._handleFrame = function _handleFrame(frame) {
 
   // Session window update
   if (frame.type === 'WINDOW_UPDATE' && frame.id === 0) {
-    if (state.version < 3.1 && state.autoSpdy31) {
```

Comment:
Just to be clear, I made additional changes that partially revert my previous change (I put the `&& state.autoSpdy31` check back in) and added the `&& state.version >= 3.1` condition to https://github.com/indutny/spdy-transport/blob/master/lib/spdy-transport/connection.js#L246 and https://github.com/indutny/spdy-transport/blob/master/lib/spdy-transport/connection.js#L475.

So now, the server doesn't send a `id=0 WINDOW_UPDATE` as soon as the client connects. Instead it waits for the client to send up a `id=0 WINDOW_UPDATE`, upgrades the connection to SPDY/3.1 and then sends the `id=0 WINDOW_UPDATE`.